generic_throttle/src/generic_throttle/generic_throttle.py

```diff
@@ -101,6 +118,19 @@ def subscriber_callback(self, data, topic_id):
                           + topic_id)
             return
 
+        try:
+            resolution = self.topics[topic_id]['resolution']
+        except KeyError:
+            resolution = None
+
+        if resolution is not None:
+            if data._type != 'sensor_msgs/Image':
+                rospy.logwarn('Resolution option is not available for ' +
```

Comment:
If the user specifies the resolution_factor option for a non-image message, the throttle will throw the warning only once when the first non-image message comes through (as the resolution option for that non-image topic will be set to `None` - line 131 - and the `rospy.logwarn` statement will not be reached any longer).